app/controlplane/internal/service/service.go

```diff
@@ -136,6 +139,75 @@ func WithLogger(logger log.Logger) NewOpt {
 	}
 }
 
+func WithEnforcer(enforcer *authz.Enforcer) NewOpt {
+	return func(s *service) {
+		s.enforcer = enforcer
+	}
+}
+
+func (s *service) authorizeResource(ctx context.Context, op *authz.Policy, resourceType authz.ResourceType, resourceID uuid.UUID) error {
+	if !rbacEnabled(ctx) {
+		return nil
+	}
+
+	// Apply RBAC
+	m := entities.CurrentMembership(ctx)
+	// check for specific resource role
+	for _, rm := range m.Resources {
+		if rm.ResourceType == resourceType && rm.ResourceID == resourceID {
+			pass, err := s.enforcer.Enforce(string(rm.Role), op)
+			if err != nil {
+				return handleUseCaseErr(err, s.log)
+			}
+			if !pass {
+				return errors.Forbidden("forbidden", "user not authorized")
+			}
+			return nil
+		}
+	}
+	return errors.Forbidden("forbidden", "user not authorized")
+}
+
+func (s *service) userHasPermissionOnProject(ctx context.Context, pUC *biz.ProjectUseCase, orgID string, pName string, policy *authz.Policy) error {
+	if !rbacEnabled(ctx) {
+		return nil
+	}
+
+	p, err := pUC.FindProjectByReference(ctx, orgID, &biz.EntityRef{Name: pName})
+	if err != nil {
+		return handleUseCaseErr(err, s.log)
+	}
+
+	return s.authorizeResource(ctx, policy, authz.ResourceTypeProject, p.ID)
+}
+
+func (s *service) visibleProjects(ctx context.Context, policy *authz.Policy) ([]uuid.UUID, error) {
+	if !rbacEnabled(ctx) {
+		return nil, nil
```

Comment:
this logic is confusing to me too, so visibleProjects return no projects if I am an admin of all the projects?

I'd rather make this method error `ProjectRBACNotNeeded` or similar so it's clear that we just don't need this check?